dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,173 @@
+    created_on: The date this entity was created.
+    created_by: The ID of the user that created this entity.
+    is_public: True to indicate if this entity is public.
+    """
+
+    name: str
+    id: int
+    parent_id: int
+    node_type: str
+    content_type: str
+    created_on: str
+    created_by: str
+    is_public: bool
+
+
+@dag(**dag_config)
+def datasets_or_projects_created_7_days() -> None:
+    """Execute a query on Snowflake and report the results to a Synapse table."""
+
+    @task
+    def get_datasets_projects_created_7_days(**context) -> List[EntityCreated]:
+        """Execute the query on Snowflake and return the results."""
+        snow_hook = SnowflakeHook(context["params"]["snowflake_conn_id"])
+        ctx = snow_hook.get_conn()
+        cs = ctx.cursor()
+        query = f"""
+        SELECT 
+            name,
+            id,
+            parent_id,
+            node_type,
+            annotations:annotations:contentType:value as content_type,
+            TO_DATE(created_on) as entity_created_date,
+            created_by,
+            is_public,
+        FROM 
+            synapse_data_warehouse.synapse.node_latest
+        WHERE 
+            (
+                node_type IN ('dataset', 'project') 
+                OR 
+                (content_type LIKE '%dataset%' AND content_type IS NOT NULL)
+            )
+        AND 
+            (
+                entity_created_date >= CURRENT_TIMESTAMP - INTERVAL '7 days' 
+            )
+        ORDER BY entity_created_date;
+        """
+        print(query)
+        cs.execute(query)
+        created_entity_df = cs.fetch_pandas_all()
+
+        entity_created = []
+        for _, row in created_entity_df.iterrows():
+            entity_created.append(
+                EntityCreated(
+                    name=row["NAME"],
+                    id=row["ID"],
+                    node_type=row["NODE_TYPE"],
+                    parent_id=row["PARENT_ID"],
+                    content_type=row["CONTENT_TYPE"],
+                    created_on=row["ENTITY_CREATED_DATE"],
+                    created_by=row["CREATED_BY"],
+                    is_public=row["IS_PUBLIC"],
+                )
+            )
+        return entity_created
+
+    @task
+    def generate_slack_message(entity_created: List[EntityCreated], **context) -> str:
+        """Generate the message to be posted to the slack channel."""
+        message = f":synapse: Datasets or projects created in the last 7 days \n\n"
+        for index, row in enumerate(entity_created):
+            if row.content_type:
+                data_type = row.content_type.strip("[] \n").strip('"')
+            else:
+                data_type = row.node_type
+            message += f"{index+1}. <https://www.synapse.org/#!Synapse:syn{row.id}|*{row.name}*> (Type: {data_type}, Created on: {row.created_on}, Created by: <https://www.synapse.org/Profile:{row.created_by}/profile|this user>, Public: {row.is_public})\n\n"
```

Comment:
Note: the result of this message is something like: 
[SMARTER](https://www.synapse.org/#!Synapse:syn64831041) (Type: project, Created on: 2025-02-20, Created by: [this user](https://www.synapse.org/Profile:3531708/profile), Public: False)

Reply:
The vast majority of created projects being exposed in the Slack [messages](https://sagebionetworks.slack.com/archives/C08F6SKK6Q3/p1740601878184249) are not public. Is this intentional? Should this only be limited to public datasets?

Reply:
hmm based on my conversation with @thomasyu888 , we just need to get all the data for the first iteration. Maybe for the next iteration, we will only look at public project.